src/OrchardCore.Modules/OrchardCore.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -391,19 +392,16 @@ private async Task GenerateContainerPathFromPattern(AutoroutePart part)
                     ContentItem = part.ContentItem
                 };
 
-                // We keep the current culture as a reference for later
-                var currentCulture = CultureInfo.CurrentUICulture;
-
-                _contentManager ??= _serviceProvider.GetRequiredService<IContentManager>();
-                var cultureAspect = await _contentManager.PopulateAspectAsync(part.ContentItem, new CultureAspect());
-
-                CultureInfo.CurrentUICulture = cultureAspect.Culture;
+                using (var cultureScope = new CultureScope(CultureInfo.CurrentUICulture))
```

Comment:
This is kind of ironic to set the CurrentUICulture to CurrentUICulture ;)

So I will just assume that here the goal is to be able to restore the previous value correctly. Kind of over-engineering to create a scope object when a try/finally will just work here.

But otherwise I have nothing against it, let's assume that it could be used somewhere else.

Reply:
Yes, i saw both, try/finally or using block

Hmm, maybe useful to have this helper class for each future case you want to render a template but not in a request context, e.g. in the background to send an email

Yes, the usage would be

    using (var cultureScope = new CultureScope(cultureAspect))

Hmm, maybe name it `CultureSwitcher` or if only for UI `UICultureSwitcher`

Reply:
Wait a minute, seems I did something wrong here ;) No need to set the CurrentUICulture in the constructor, seems I forgot to delete from the constructor